### PR TITLE
strictGet method and tests

### PR DESCRIPTION
**Pull request by jskelcy**
There are a lot of tests, let me know if I need to add this into more tests

### CODE REVIEW

File: errors.js [hunk 1]
```diff
@@ -25,6 +25,12 @@ var DatacenterRequired = TypedError({
         'SUGGESTED FIX: update the `fetchConfig()` callsite.\n'
 });
 
+var NonexistantKeyPath = TypedError({
+    type: 'nonexistant.key.path',
```
Comment (Raynos) [inline errors.js]: `zero-config.nonexistant.key-path`

File: errors.js [hunk 2]
```diff
@@ -25,6 +25,12 @@ var DatacenterRequired = TypedError({
...
+    message: 'attempting to get a nonexistant keyPath.\n' +
```
Comment (Raynos) [inline errors.js]: message: 'attempted to get a key in ZeroConfig that does not exist.\n' +
   'Expected the key {keyPath} to have a defined value.\n' +
   'SUGGESTED FIX: Update your configuration files to define the key.\n'

File: errors.js [hunk 3]
```diff
@@ -25,6 +25,12 @@ var DatacenterRequired = TypedError({
...
+    'SUGGESTED FIX: add keyPath and value to config`'
```
Comment (Raynos) [inline errors.js]: `keyPath: null`

File: config-wrapper.js [hunk 4]
```diff
@@ -23,6 +23,21 @@ function ConfigWrapper(configObject) {
...
+    function configuredGet(keyPath){
+        if (!keyPath){
+            return getKey();
+        }
+
+        var value = getKey(keyPath);
+        var strictMode = !loose;
+
+        if (value === undefined && strictMode) {
+            throw errors.NonexistantKeyPath();
```
Comment (Raynos) [inline config-wrapper.js]: throw errors.NonexistantKeyPath({ keyPath: keyPath });

**Comment by Raynos** [comment]
lgtm. @jskelcy #shipit

**Comment by rf** [comment]
@jskelcy why haven't you rebased this yet damn

**Pull request closed without merging**